File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        l , r = 0, len(numbers)-1
        res =[]
        while l < r:

            twosum = numbers[l]+numbers[r]

            if twosum > target:

                r-=1
            elif twosum<target:
                l+=1
            else:
                res.append(l+1)
                res.append(r+1)
                l+=1
                r-=1
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {


     HashMap<Integer, Integer> map = new HashMap<>();



     for( int i =0;i<nums.length;i++)
     {
        
        if(map.containsKey(target-nums[i]))
        {
            int k = map.get(target-nums[i]);
            return new int[] {k,i};
        }
        else
            map.put(nums[i],i);
     }  
     return new int[] {0,0};
    }
}
